### PR TITLE
Update tests to use node:test, remove tap

### PR DESCRIPTION
**Pull request by bencoder**
Updates tests to use node:test
Remove tap
Converted tests to async/await style over callbacks and removed `t.plan(x)` statements where appropriate

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by Uzlopak** [comment]
Yeah the windows fails. Maybe simplest solution: change file extensions of all test files from `.js` to `.test.js`

**Review by gurgunday — CHANGES_REQUESTED** [review]
Hi, thanks a lot for the PR!

Please:

1) Install `c8`
2) Change the `test:unit` command to `c8 --100 node --test`
3) As @Uzlopak said, for consistency, it would be great if you could add `.test.` to the test files

**Review by dancastillo — COMMENTED** [review]
Can you add back the `t.plan` to the tests `node:test` does support test count assertions [ref](https://nodejs.org/api/test.html#contextplancount)

**Comment by bencoder** [comment]
@dancastillo I've removed them where they are not adding anything - they only really make sense when you have assertions in callbacks or conditional statements - everywhere I've removed them has been where I've changed to using `await` so the assertions will always run and setting `t.plan` just seems like extra friction. 

If they should be a standard for fastify tests though then I can add them back if necessary

**Comment by dancastillo** [comment]
@bencoder That makes sense to me! As for the standard with Fastify tests, I believe we have them in place for most of our tests but like you mention likely for callbacks or conditionals

I'll defer it to @Uzlopak or @gurgunday to confirm whether it's a strict requirement or is okay

**Comment by jsumners** [comment]
I would reserve plans for tests of asynchronous operations.

**Review by gurgunday — APPROVED** [review]
lgtm

File: test/access.test.js [hunk 1]
```diff
@@ -93,35 +90,32 @@ t.test('Get all data that we set in session', t => {
...
-      }
-    }, (error, response) => {
-      t.error(error)
-      t.same(JSON.parse(response.payload), {
-        data1: { some: 'data' },
-        data2: { some: 'data' }
-      })
-    })
+  })
+  t.assert.ifError(postResponse.error)
```
Comment (dancastillo) [inline test/access.test.js:100]:
```suggestion
  t.assert.ok(postResponse)
```

File: test/access.test.js [hunk 2]
```diff
@@ -93,35 +90,32 @@ t.test('Get all data that we set in session', t => {
...
+  t.assert.ok(postResponse.headers['set-cookie'])
+
+  const getResponse = await fastify.inject({
+    method: 'GET',
+    url: '/',
+    headers: {
+      cookie: postResponse.headers['set-cookie']
+    }
+  })
+  t.assert.ifError(getResponse.error)
```
Comment (dancastillo) [inline test/access.test.js:111]:
```suggestion
  t.assert.ok(getResponse)
```

**Review by dancastillo — COMMENTED** [review]
Looks like there is no `error` defined on `await fastify.inject(...)` response [ref](https://github.com/fastify/light-my-request/blob/master/types/index.d.ts#L78-L93)

So this below will never fail. 
```js
t.assert.ifError(response.error)
```

Can we update all similar asserts to below to validate we received a truthy response
```js
  t.assert.ok(response)
```

**Review by dancastillo — APPROVED** [review]
LGTM

**Pull request merged**